packages/core/src/entity/Collection.ts

```diff
@@ -204,6 +204,14 @@ export class Collection<T extends object, O extends object = object> extends Arr
     return super.isEmpty();
   }
 
+  /**
+   * @inheritDoc
+   */
+  slice(start = 0, end?: number): T[]  {
```

Comment:
no need for the default here, its enough to have it in the base method
```suggestion
  slice(start?: number, end?: number): T[]  {
```